### ISSUE TITLE
DPL-568 Bioscan - Labware creator to create Lysate plate

### ISSUE THREAD

**Issue by andrewsparkes** [sanger/limber#1196]
**User story**
As PSD we will need a new Labware creator to create the Lysate plate in the Insect Lysate Prep pipeline. This will need to determine the randomised locations of where the two PCR controls will be positioned and create those Control samples in Sequencescape. The Beckman robot method will then call an API (separate story) to fetch those control well locations and perform the transfers.

**Who are the primary contacts for this story**
Scott T
Emma B

**Who is the nominated tester for UAT**
TBC

**Acceptance criteria**
To be considered successful the solution must allow:
- [x] labware creator makes the LBSN-96 Lysate plate (with addition of controls) from a stock LILYS-96 Stock plate source
- [x] labware creator needs to create the 2 control samples, using parameters stored in Limber config on the plate purpose
- [x] control locations are to be randomised on the plate (subject to the Beckman limitation described below)
- [x] the bioscan integration test should run using the creator

**Dependencies**
This story is blocked by the following dependencies:
Development and SATs for the Beckman robot (completed early April 2023)

**References**
This story has a non-blocking relationship with:
Integration test - https://github.com/sanger/General-Backlog-Items/issues/233
Limber pipeline configuration - https://github.com/sanger/sequencescape/issues/3703
Sequencescape pipeline configuration - https://github.com/sanger/sequencescape/issues/3703
Limber bed verification for Stock plate location - https://github.com/sanger/limber/issues/1198
Limber bed verification for Stock to Lysate plate - https://github.com/sanger/limber/issues/1199

**Additional context**
This is the part of the pipeline where the stock insect plate is transferred into the lysate plate and positive and negative PCR controls are added.
An insect sample can be lost in order to randomise the control placements. Typically we expect the H12 location for the positive control to NOT have an insect sample in it. [This part about H12 obsolete, Beckman can cope with both controls being randomised other than restriction mentioned below]
See diagram [here](https://lucid.app/lucidchart/b21a6020-df09-4be1-af69-26403ed467a1/edit?viewport_loc=1860%2C1619%2C1885%2C1065%2Cwi47cHrbc6ZW&invitationId=inv_44a43cc4-479b-4e19-924c-2a0182f4255c)

From Scott 27th March re: Beckman limitiation:
Beckman are able to handle the randomisation of both positive and negative control coordinates with one exception:
Where positive_control= H1 AND negative_control = G1.
Any other combination works (including either one of these conditions being true)

**Comment by andrewsparkes** [comment]
Lot pf problems with this story around the use of the Sequencescape version 2 JSON API. Eventually got it working but had to make a few changes to the JSON API relationships in Limber. Mocking those interactions for the tests was also not straight forward.

**Comment by andrewsparkes** [comment]
At UAT Scott has mentioned an additional restriction, that neither PCR control should be placed in H12, as H12 may be used to hold the negative Lysate control. Will confirm requirements and write up as separate stories.
